tests/merkle-tree.ts

```diff
@@ -238,3 +240,67 @@ export function hashLeaves(leaves: Buffer[]): Buffer {
     }
     return nodes[0];
 }
+
+
+export type AssetProof = {
+    root: string,
+    hash: string,
+    proof: string[],
+}
+
+export type NodeView = {
+    hash: string,
+    level: number,
+    index: number
+}
+
+// Very annoying, since last element in proof is actually root
```

Comment:
Correct, very annoying. We should fix the server to not do this lol

Reply:
Unless comment is out of date

Reply:
Comment is out of date haha

Reply:
I think it would be good to have an API call that looks things up by user (can use to make a second query to one of these other endpoints)

Reply:
Addressed in latest commit